x-pack/plugins/viz_editor/public/pseudo_plugins/essql_datasource_plugin/essql_datasource_plugin.tsx

```diff
@@ -15,13 +15,15 @@ function DataPanel(props: DatasourcePanelProps<VisModel>) {
   const [text, updateText] = useState(visModel.datasource ? visModel.datasource.meta.sql : '');
 
   const updateDatasource = async () => {
-    const resultColumns: Array<{ name: string; type: string }> = await kfetch({
-      pathname: '/api/viz_editor/sqlfields',
-      method: 'POST',
-      body: JSON.stringify({
-        sql: text,
-      }),
-    });
+    const resultColumns: Array<{ name: string; type: 'number' | 'string' | 'date' }> = await kfetch(
```

Comment:
Could you use a named type instead of a union here? Or is it just 'string'?

Reply:
This comes from the essql server function, I connected it on a type level and use `FieldType` here.